flink-core/src/main/java/org/apache/flink/configuration/Configuration.java

```diff
@@ -843,11 +843,14 @@ private float convertToFloat(Object o, float defaultValue) {
 		}
 		else if (o.getClass() == Double.class) {
 			double value = ((Double) o);
-			if (value <= Float.MAX_VALUE && value >= Float.MIN_VALUE) {
-				return (float) value;
-			} else {
+			if (value < -Float.MAX_VALUE
```

Comment:
The current checks around `0` can be combined. Can we simplify this to two ranges, `-max` to `-min` and `min` to `max` (all inclusive), reverting the order of the if-else?